### PR TITLE
fix(MegaMenu): explicity set focus to HeaderMenu L0 item when clicked

### PR DESCRIPTION
**Pull request by annawen1**
### Related Ticket(s)

[MM] My IBM maintaining focus when switching to mega menu #3617
[HP] [MM] multiple menus open at once #3731

### Description

The `<OverflowMenu />` component that we are using for the profile menu has logic to trap the focus when user clicks outside of the component. This affects the megamenu when user opens the profile menu, then clicks on the L0 nav item. The focus is trapped in the `<OverflowMenu />` which affects the `onBlur` event triggered used when opening another L0 nav item.

This PR forces focus on the L0 nav item when it is clicked.

**BEFORE:**
Aug-26-2020 16-17-05

**AFTER:**
Aug-26-2020 16-19-33

### Changelog

**Changed**

- force focus on the `HeaderMenu` L0 item

### CODE REVIEW

File: packages/react/src/components/carbon-components-react/UIShell/HeaderMenu.js [hunk 1]
```diff
@@ -125,6 +126,7 @@ class HeaderMenu extends React.Component {
       `${prefix}--masthead__megamenu__category-headline`,
       `${prefix}--masthead__megamenu__category-group`,
       `${prefix}--masthead__megamenu__view-all-cta`,
+      `${prefix}--masthead__megamenu__l0-nav`,
```
Comment (annawen1) [inline packages/react/src/components/carbon-components-react/UIShell/HeaderMenu.js:129]: adding this here so when user clicks on another L0 menu item, the overlay doesn't flash

**Comment by jeffchew** [comment]
@annawen1 Looks good! 

I'm thinking through this, and we are already set to go as-is with what we have currently tonight, so I'm thinking of having this retargeted to a release branch for a patch release instead, so let's hold on merging this for now.

**Comment by jeffchew** [comment]
@annawen1 one other thought I have, perhaps we can target `master` for now so we can test around this on `canary`, and if we do a patch we can always cherry pick this commit. Thoughts?

**Comment by annawen1** [comment]
> @annawen1 one other thought I have, perhaps we can target `master` for now so we can test around this on `canary`, and if we do a patch we can always cherry pick this commit. Thoughts?

@jeffchew yeah sure that works

**Pull request merged**